gateway/sqlite.go

```diff
@@ -26,20 +26,85 @@ var (
 	errDBClosed         = errors.New("error gateway is closed")
 )
 
+// Node defines a lorawan node device.
+type gatewayNode struct {
+	AppSKey []byte
+	NwkSKey []byte
+	AppKey  []byte
+
+	Addr              []byte
+	DevEui            []byte
+	NodeName          string
+	MinUplinkInterval float64
+
+	DecoderPath string
+	JoinType    string
+
+	FCntDown  uint16
+	FPort     byte     // 11 for downlinks, only required when frame payload exists.
+	Downlinks [][]byte // 12? list of downlink frame payloads to send
+}
+
+const (
+	devEUIDBKey            = "devEui"
+	appSDBKey              = "appSKey"
+	nwkSDBKey              = "nwkSKey"
+	devAddrDBKey           = "devAddr"
+	fCntDownDBKey          = "fCntDown"
+	nodeNameDBKey          = "nodeName"
+	minUplinkIntervalDBKey = "minUplinkInterval"
+	appKeyDBKey            = "appKey"
+	decoderPathDBKey       = "decoderPath"
+	joinTypeDBKey          = "joinType"
+	fPortDBKey             = "fPort"
+	isRegisteredDBKey      = "isRegistered"
```

Comment:
where are these fields used?

Reply:
these fields are placeholders, meant to be used instead of `gatewayNode`. equivalent code is currently used in `join.go`, `uplink.go` and `downlink.go`